src/loadMessages.js

```diff
@@ -0,0 +1,11 @@
+export default function loadMessages(lang) {
+  const wrapped = import(`./messages/${lang}`)
+
+  if (typeof window !== 'undefined') {
+    return wrapped.then(msgs => msgs.default)
+  }
+
+  /* eslint-disable global-require, import/no-dynamic-require, prefer-template */
+  // return Promise.resolve(require('./messages/' + lang).default)
+  return __webpack_require__(wrapped.resolve()).default
+}
```

Comment:
ur right, that code will work. it doesnt need to have a props arg passed. i guess i made the babel plugin more flexible than i thought. that said, it was never intended for people to call methods like `resolve` directly. thats why i was suggesting to just do it manually using resolveWeak. refresh the page and see the example i gave above.

...as for chunks, this isnt handled by webpack flush chunks, since we only handle components currently. u should be able to do this though:

```js
wrapped.chunkName()
```

and then add that to the array of chunkNames u pass to flushChunks(stats, { chunkNames }).

so in this case, it's actually a great idea to use the babel plugin, rather than do resolveWeak directly. I stand corrected!

So here's the new flushing code:

```js
const chunkNames = flushChunknames()
chunkNames.push(wrapped.chunkName())
const { scripts } = flushChunks(stats, { chunkNames })
```

That should work.